apps/mysql.go

```diff
@@ -15,13 +15,65 @@
 package apps
 
 import (
-	"fmt"
-	"strings"
-
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+	"fmt"
+	"strings"
 )
 
+type MetricsReceiverMySQL struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url"`
+
+	Password string `yaml:"password" validate:"omitempty"`
+
+	Username string `yaml:"username" validate:"omitempty"`
+}
+
+const defaultEndpoint = "localhost:3306"
```

Comment:
This receiver is another example where you're using the `url` validator and the default value is not a URL. We need to be very clear and explicit with how users should configure this - are there valid URLs that can go here? What does a Unix socket path look like? etc.

Reply:
So this is going to be `host:port` as well. I've made an upstream PR to update it to add an additional parameter to enable different protocols.

Reply:
Is there a link to the upstream PR?

Reply:
I don't think we should merge this without Unix socket support

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6138/files

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6138 was merged.